lib/cylc/config.py

```diff
@@ -1504,55 +1498,105 @@ def generate_taskdefs(self, orig_expr, left_nodes, right, section, seq,
+            for item in lst:
+                if type(item) is list:
+                    ret.append(recursive_replace(item, obj, replacement))
+                elif item == obj:
+                    ret.append(replacement)
+                else:
+                    ret.append(item)
+            return ret
+
+        trigers = set([])
         for left in left_nodes:
             # (GraphNodeError checked above)
-            cycle_point = None
             lnode = graphnode(left, base_interval=base_interval)
             ltaskdef = self.taskdefs[lnode.name]
 
+            # Determine intercycle offsets.
+            abs_cycle_point = None
+            cycle_point_offset = None
             if lnode.offset_is_from_ict:
                 first_point = get_point_relative(
                     lnode.offset_string, self.initial_point)
                 last_point = seq.get_stop_point()
+                abs_cycle_point = first_point
                 if last_point is None:
                     # This dependency persists for the whole suite run.
                     ltaskdef.intercycle_offsets.add(
                         (None, seq))
                 else:
                     ltaskdef.intercycle_offsets.add(
                         (str(-(last_point - first_point)), seq))
-                cycle_point = first_point
+            elif lnode.is_absolute:
+                abs_cycle_point = lnode.offset_string
             elif lnode.intercycle:
                 if lnode.offset_is_irregular:
                     offset_tuple = (lnode.offset_string, seq)
                 else:
                     offset_tuple = (lnode.offset_string, None)
                 ltaskdef.intercycle_offsets.add(offset_tuple)
+                cycle_point_offset = lnode.offset_string
 
-            trig = TaskTrigger(
-                lnode.name, lnode.output, lnode.offset_string, cycle_point,
-                suicide, self.cfg['runtime'][lnode.name]['outputs'],
-                base_interval)
+            # Check for message trigger offsets.
+            outputs = self.cfg['runtime'][lnode.name]['outputs']
+            for message in outputs.values():
+                if BCOMPAT_MSG_RE_C6.match(message):
+                    raise SuiteConfigError(
+                        'ERROR: Message trigger offsets are obsolete.')
 
-            # Use fully qualified name for trigger expression label
-            # (task name is not unique, e.g.: "F | F:fail => G").
-            label = self.get_conditional_label(left)
-            ctrig[label] = trig
-            cname[label] = lnode.name
+            # Qualifier.
+            if outputs and lnode.output in outputs:
+                # Task message.
+                qualifier = outputs[lnode.output]
+            else:
+                # Built-in qualifier.
+                if lnode.output:
+                    qualifier = TaskTrigger.get_trigger_name(lnode.output)
+                else:
+                    qualifier = TASK_OUTPUT_SUCCEEDED
```

Comment:
```python
            if outputs and lnode.output in outputs:
                # Task message.
                qualifier = outputs[lnode.output]
            elif lnode.output:
                # Built-in qualifier.
                qualifier = TaskTrigger.get_trigger_name(lnode.output)
            else:
                qualifier = TASK_OUTPUT_SUCCEEDED
```

A slightly better style? (This lines up all the assignment statements of `qualifier`.)